test/Rings-conformance-tests.jl

```diff
@@ -21,6 +21,113 @@ function equality(a::T, b::T) where T <: AbstractAlgebra.NCRingElement
    end
 end
 
+function test_mutating_op_like_zero(f::Function, f!::Function, a)
+  A = deepcopy(a)
+  a = f!(a)
+  @test equality(a, f(A))
+  a = deepcopy(A)
```

Comment:
(Why) is this needed?
```suggestion
```

Reply:
I want to make sure that these test functions in no case mutate their arguments. But with your suggestion in https://github.com/Nemocas/AbstractAlgebra.jl/pull/1816/files#r1782733905 to switch the names, this is not needed. I'll adapt